File: palindrome-number/palindrome-number.cpp
// https://leetcode.com/problems/palindrome-number/

#include <iostream>
#include <string>
#include <sstream>


class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0)
            return false;

        std::string string_x = "";
        std::stringstream stream;
        stream << x;
        stream >> string_x;

        int len_digits = string_x.size();
        int i = 0;
        int j = len_digits - 1;
        bool is_palindrome = true;
        while( i < len_digits ) {
            if ( string_x[i] != string_x[j] ) {
                is_palindrome = false;
                break;
            }

            i += 1;
            j -= 1;
        }

        return is_palindrome;
    }
};


int
main(int argc, char* argv[]) {
    bool result = Solution().isPalindrome( std::stoi(argv[1]) );
    std::cout << result << "\n";
}

File: palindrome-number/palindrome-number.py
# https://leetcode.com/problems/palindrome-number/
"""
Given an integer x, return true if x is a
palindrome, and false otherwise.

 

Example 1:

Input: x = 121
Output: true
Explanation: 121 reads as 121 from left to right and from right to left.

Example 2:

Input: x = -121
Output: false
Explanation: From left to right, it reads -121. From right to left, it becomes 121-. Therefore it is not a palindrome.

Example 3:

Input: x = 10
Output: false
Explanation: Reads 01 from right to left. Therefore it is not a palindrome.

 
Constraints:

    -2^31 <= x <= 2^31 - 1

"""

import unittest
import copy


class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        array_x = [
            c
            for c in str(x)
        ]
        comparison_array = copy.copy(array_x)
        comparison_array.reverse()

        return comparison_array == array_x


class Test__Solution(unittest.TestCase):

    def test__first_input(self):
        assert Solution().isPalindrome(x=121) == True

    def test__second_input(self):
        assert Solution().isPalindrome(x=-121) == False

    def test__third_input(self):
        assert Solution().isPalindrome(x=10) == False


if __name__ == '__main__':
    unittest.main()